tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1431,6 +1447,8 @@ public function testPrimaryKeyAutoIncrement() : void
         $this->connection->insert('test_pk_auto_increment', ['text' => '1']);
 
         $query = $this->connection->query('SELECT id FROM test_pk_auto_increment WHERE text = \'1\'');
+        assert($query instanceof Statement);
```

Comment:
Change the `Doctrine\DBAL\Connection::query()` return type?

Reply:
If possible, yes. Wherever a DBAL Connection returns a DBAL Statement, it should be clear from the API because regardless of the fact that the Driver and DBAL layer share the same interfaces (which is bad), they are not equal (that's why it's bad).

Reply:
Cannot be done due to https://github.com/doctrine/dbal/issues/3591.